### PR TITLE
Hunterize libigl, attempt 2

### PR DESCRIPTION
**Pull request by rbsheth**
Original PR with missing branch here: #1202 

This PR adds basic support for the Hunter package manager in libigl. Because libigl is already following modern CMake practices, the changes needed to make this package Hunter-ready are minimal.

Changes:
1) Add `HunterGate.cmake` and corresponding CMake code in `CMakeLists.txt`
2) Pull `Eigen` from Hunter
3) Add `HUNTER_ENABLED` flag and default to `OFF` to preserve current behavior.

Once this change is merged, a new (minor?) GitHub release is needed to generate a source tarball that Hunter can pull in.

#### Check all that apply (change to `[x]`)
- [x] All changes meet [libigl style-guidelines](https://libigl.github.io/style-guidelines/).
- [ ] Adds new .cpp file.
- [ ] Adds corresponding unit test.
- [ ] Adds corresponding python binding.
- [x] This is a minor change.

### CODE REVIEW

File: cmake/libigl.cmake [hunk 1]
```diff
@@ -80,6 +80,10 @@ if(MSVC)
   # Enable parallel compilation for Visual Studio
   target_compile_options(igl_common INTERFACE /MP /bigobj)
   target_compile_definitions(igl_common INTERFACE -DNOMINMAX)
+  target_compile_options(igl_common INTERFACE /Zc:__cplusplus)
```
Comment (jdumas) [inline cmake/libigl.cmake]: Can you explain why this is needed? Maybe it's worth discussing in a separate PR.

File: cmake/libigl.cmake [hunk 2]
```diff
@@ -80,6 +80,10 @@ if(MSVC)
...
+  if(LIBIGL_USE_STATIC_LIBRARY)
+    target_compile_definitions(igl_common INTERFACE -D_USE_MATH_DEFINES)
```
Comment (jdumas) [inline cmake/libigl.cmake]: This shouldn't be necessary, as we take care to not use `M_PI` directly in libigl. Is there any reason to enable this flag?

File: cmake/libigl.cmake (hunk 1 above)
Comment (rbsheth) [reply cmake/libigl.cmake]: Background on the flag: https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/

I could swear there was some code gated on the `__cplusplus` version (which I subsequently used here for porting `master`: https://github.com/hunter-packages/libigl/commit/5f2a02d3eec78b94b701907f66ffbaafd0f6d0a1#diff-379356ad4dff6b985fbb126dc1a6fd32R11) but I can't find it. I can remove it for this PR.

File: cmake/libigl.cmake (hunk 2 above)
Comment (rbsheth) [reply cmake/libigl.cmake]: Same issue, perhaps it was fixed earlier but in my experiment `M_PI` had crept in somehow. I can remove this too.

Comment (jdumas) [reply cmake/libigl.cmake]: Yes, we should instead remove all usage of `M_PI` from libigl's code instead, without requiring additional compilation flags from the user.

File: cmake/libigl.cmake (hunk 1 above)
Comment (jdumas) [reply cmake/libigl.cmake]: Libigl's minimum requirement in terms of C++ version is C++11. We don't seem to be relying on the `__cplusplus` macro to induce some specific behavior, so I don't see the hurt in not specifying this flag.

**Comment by rbsheth** [comment]
@jdumas Any ideas on this Travis failure? https://travis-ci.org/libigl/libigl/jobs/550530668

Looks like `ccache` is missing?

**Comment by jdumas** [comment]
Yes. I'm waiting for another maintainer to approve #1239 (I can't merge it myself) which fixes the CI issues. Once this is done we can merge this PR.

**Comment by jdumas** [comment]
@rbsheth I'm trying to merge the tip of the `dev` branch with your PR branch to trigger a new build on Travis, but it doesn't look like you've given us write access to the branch when you open your PR. Could you see to it?

**Comment by rbsheth** [comment]
@jdumas Invited you to collaborate

**Comment by jdumas** [comment]
@rbsheth do you mind if I squash and merge instead of just merging? Or will it affect your fork in hunter for future versions?

**Comment by rbsheth** [comment]
@jdumas Squash merge should be fine, I can work with that for Hunter.

**Pull request merged**